### PR TITLE
Add button autofocus prop to confirm modal

### PR DESCRIPTION
**Pull request by ghost**
@picodoth

### CODE REVIEW

File: components/modal/confirm.tsx [hunk 1]
```diff
@@ -9,6 +9,7 @@ import { getConfirmLocale } from './locale';
 interface ConfirmDialogProps extends ModalFuncProps {
   afterClose?: () => void;
   close: (...args: any[]) => void;
+  autoFocusBtn: 'ok' | 'cancel';
```
Comment (afc163) [inline components/modal/confirm.tsx]: Consider https://github.com/ant-design/ant-design/pull/11055 and we can set `autoFocusBtn` to `null|false|undefined` for no focused button.

Comment (afc163) [reply components/modal/confirm.tsx]: autoFocusBtn => autoFocusButton

File: components/modal/confirm.tsx [hunk 2]
```diff
@@ -28,6 +29,7 @@ const ConfirmDialog = (props: ConfirmDialogProps) => {
   const okText = props.okText ||
     (okCancel ? runtimeLocale.okText : runtimeLocale.justOkText);
   const cancelText = props.cancelText || runtimeLocale.cancelText;
+  const focus = props.autoFocusBtn || 'ok';
```
Comment (afc163) [inline components/modal/confirm.tsx]: Don't rename autoFocusBtn;

**Comment by afc163** [comment]
ci failed

**Comment by erwin-k** [comment]
@afc163 Done. 
Don't know about ci failure, errors don't seem related to these changes at all.

**Comment by afc163** [comment]
Don't forget update API documentation.

File: components/modal/confirm.tsx [hunk 3]
```diff
@@ -9,6 +9,7 @@ import { getConfirmLocale } from './locale';
 interface ConfirmDialogProps extends ModalFuncProps {
   afterClose?: () => void;
   close: (...args: any[]) => void;
+  autoFocusButton?: any;
```
Comment (abenhamdine) [inline components/modal/confirm.tsx]: Why not: 
`autoFocusButton?: null | 'ok' | 'cancel'` ?

**Review by abenhamdine — CHANGES_REQUESTED** [review]

**Comment by erwin-k** [comment]
@afc163 Can I add this feature to regular modals aswell?

**Comment by afc163** [comment]
@erwin-k That is cool

**Comment by erwin-k** [comment]
@afc163 In the end didn't do it for Modal component cause it'd only focus on first open.

File: components/modal/confirm.tsx [hunk 4]
```diff
@@ -28,6 +29,7 @@ const ConfirmDialog = (props: ConfirmDialogProps) => {
   const okText = props.okText ||
     (okCancel ? runtimeLocale.okText : runtimeLocale.justOkText);
   const cancelText = props.cancelText || runtimeLocale.cancelText;
+  const autoFocusButton = props.autoFocusButton === null ? false : props.autoFocusButton || 'ok'
```
Comment (picodoth) [inline components/modal/confirm.tsx]: why do we need this line? We can simply use props.autoFocusButton directly. 
```
const { autoFocusButton } = this.props;
```

File: components/modal/index.en-US.md [hunk 5]
```diff
@@ -62,6 +62,7 @@ The properties of the object are follows:
 
 | Property | Description | Type | Default |
 | -------- | ----------- | ---- | ------- |
+| autoFocusButton | Specify which button to autofocus | null\|string: `ok` `cancel` | `ok` |
```
Comment (picodoth) [inline components/modal/index.en-US.md]: can you add something like this to the corresponding place in components/modal/index.zh-CN.md

```
| autoFocusButton | 指定自动获得焦点的按钮 | null\|string: `ok` `cancel` | `ok` |
```

File: components/modal/confirm.tsx (hunk 4 above)
Comment (erwin-k) [reply components/modal/confirm.tsx]: Wouldn't that give 'ok' when null is passed? Passing null serves to disable autofocus in this case.

**Comment by afc163** [comment]
ci is still broken.

**Comment by erwin-k** [comment]
@afc163 I don't know about the failures, looks like it has nothing to do with these files. I have no time to investigate all that just for a pull request.

**Comment by afc163** [comment]
Try rebase `feature-3.9.0`

**Comment by afc163** [comment]
Try rebase `feature` branch now.

**Comment by aaronplanell** [comment]
Sorry @afc163 , how can I rebase the `feature` branch if I cannot download the code?

**Comment by aaronplanell** [comment]
Hello @afc163 ,

Maybe this is one of the problems with the PR of the thread: 
1. I fork the Ant Design project.
2. I switch to the `feature` branch.
3. I run `nmp install`
4. I run `npm t`

There is a snapshot that fails:
```
Summary of all failing tests
 FAIL  components/tree/__tests__/demo.test.js
  ● renders ./components/tree/demo/basic.md correctly

    expect(value).toMatchSnapshot()

    Received value does not match stored snapshot "renders ./components/tree/demo/basic.md correctly 1".

    - Snapshot
    + Received
```

Maybe this is the original problem.

**Pull request merged**